### PR TITLE
Sofast Configuration class

### PR DESCRIPTION
**Pull request by braden6521**
- Added Sofast Configuration class. This is the first implementation. Currently it can return measurement statistics and visualize the setup. 
- Applies to both Sofast Fringe and Fixed.
- Removed old visualize_setup script as that functionality is now part of SofastConfiguration.

### CODE REVIEW

File: doc/source/library_reference/app/sofast/config.rst [hunk 1]
```diff
@@ -98,16 +98,6 @@ opencsp.app.sofast.lib.PatternSofastFixed
    :undoc-members:
    :show-inheritance:
 
-opencsp.app.sofast.lib.visualize_setup
```
Comment (e10harvey) [inline doc/source/library_reference/app/sofast/config.rst:101]: I see that this function still exists as a public API. Why should we not include it in our online documentation here?

Comment (braden6521) [reply doc/source/library_reference/app/sofast/config.rst:101]: @e10harvey, Thanks for the review and the comment; what do you mean by "public API?"

Comment (e10harvey) [reply doc/source/library_reference/app/sofast/config.rst:101]: By "public API" I mean a public application programming interface, in this case a public python function or method that is intended to be called by end-users of OpenCSP. My understanding is that python doesn't prevent end users from calling private and protected methods/functions, but rather, relies on using `__` and `_` to denote that the method/function is private or protected but python still lets anyone call these functions. So, functions/methods that do not begin with `__` or `_` are considered public and are considered fair game for users to call.

Comment (e10harvey) [reply doc/source/library_reference/app/sofast/config.rst:101]: @braden6521

Comment (braden6521) [reply doc/source/library_reference/app/sofast/config.rst:101]: @e10harvey, I think I'm tracking. The "sofast.lib.visualize_setup()" script was deleted as part of this PR, thus I removed it from config.rst. However, it now exists as a method in sofast.lib.SofastConfiguration.visualize_setup(). To not lose information on this function, I've added the sofast.lib.SofastConfiguration() class to config.rst.
**NOTE**: There are still some functions in sofast.lib that aren't in config.rst; that's probably a topic for another PR.

Comment (e10harvey) [reply doc/source/library_reference/app/sofast/config.rst:101]: Yes, thank you

**Review by e10harvey — APPROVED** [review]

**Pull request merged**